src/python/pants/engine/internals/parser.py

```diff
@@ -101,7 +103,24 @@ def parse(
         except NameError as e:
             valid_symbols = sorted(s for s in global_symbols.keys() if s != "__builtins__")
             original = e.args[0].capitalize()
-            raise ParseError(f"{original}.\n\nAll registered symbols: {valid_symbols}")
+            help_str = (
+                "If you expect to see more symbols activated in the below list,"
+                f" refer to {docs_url('enabling_backends')} for all available"
```

Comment:
Oh this needs to be `enabling-backends` with a dash.

Reply:
> Oh this needs to be `enabling-backends` with a dash.

This is a pretty easy mistake to make. It would be good to have some kind of checking, probably in `docs_url`, that you're not using the wrong version of `'-', '_'` and accidentally making a broken link.

Reply:
@gshuflin Should I open an issue for that? I could probably get to it early next week.

Also, for some reason your other comment about the multiline isn't coming up for me to comment on (so apologies for out of place-ness) -- I tried that at first, and ran into weird issues that I couldn't resolve with indentation (while still keeping within the formatting requirements) and got the implicit concatenation suggestion from Eric. Do you happen to know of a library/technique to resolve this? I couldn't manage to get it to work with either the `textwrap` or `inspect` libraries.

Reply:
Sometimes it is an underscore, like `reference-python_library` :/

We'd benefit from an automated way to audit all of `./pants help` for broken links, along with the docsite for broken links. Maybe it's as simple as `./pants help-all`, searching for all `https://`, and using something like requests or curl to check if it's broken?

Reply:
> I tried that at first, and ran into weird issues that I couldn't resolve with indentation

Yeah this is the challenge with multiline strings - they'll use the same indentation of your source code. As clunky as it is, I think implicit string concatenation is the way to go here.

Reply:
> @gshuflin Should I open an issue for that? I could probably get to it early next week.
> 
> Also, for some reason your other comment about the multiline isn't coming up for me to comment on (so apologies for out of place-ness) -- I tried that at first, and ran into weird issues that I couldn't resolve with indentation (while still keeping within the formatting requirements) and got the implicit concatenation suggestion from Eric. Do you happen to know of a library/technique to resolve this? I couldn't manage to get it to work with either the `textwrap` or `inspect` libraries. Thanks for the input!

Yeah, I actually deleted that comment because I thought better of it. Like Eric says, implicit string concatenation is clunky but it's probably the best option here.

Reply:
> @gshuflin Should I open an issue for that? I could probably get to it early next week.
> 
> Also, for some reason your other comment about the multiline isn't coming up for me to comment on (so apologies for out of place-ness) -- I tried that at first, and ran into weird issues that I couldn't resolve with indentation (while still keeping within the formatting requirements) and got the implicit concatenation suggestion from Eric. Do you happen to know of a library/technique to resolve this? I couldn't manage to get it to work with either the `textwrap` or `inspect` libraries. Thanks for the input!

It's rarely a bad idea to make an issue. I'm not sure what the right fix is here. Maybe there are never any underscores in our docs URLs and we can throw an exception if we detect one, but I'd have to check and be sure that's the case first. Maybe it's not worth fixing at all. In any case it's outside the scope of this PR.